libbpf-tools/tcprtt.bpf.c

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2021 Wenbo Zhang
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_endian.h>
+#include "tcprtt.h"
+#include "bits.bpf.h"
+#include "maps.bpf.h"
+
+const volatile bool targ_laddr_hist = false;
+const volatile bool targ_raddr_hist = false;
+const volatile bool targ_show_ext = false;
+const volatile __u16 targ_sport = 0;
+const volatile __u16 targ_dport = 0;
+const volatile __u32 targ_saddr = 0;
+const volatile __u32 targ_daddr = 0;
+const volatile bool targ_ms = false;
+
+#define MAX_ENTRIES	10240
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_ENTRIES);
+	__type(key, u64);
+	__type(value, struct hist);
+} hists SEC(".maps");
+
+static struct hist zero;
+
+SEC("fentry/tcp_rcv_established")
+int BPF_PROG(tcp_rcv, struct sock *sk)
+{
+	const struct inet_sock *inet = (struct inet_sock *)(sk);
+	struct tcp_sock *ts;
+	struct hist *histp;
+	u64 key, slot;
+	u32 srtt;
+
+	if (targ_sport && targ_sport != inet->inet_sport)
```

Comment:
I think we need `bpf_ntohs` here.

Reply:
Indeed, it is comparing different orderings.  Did it work in your test?

All these global targ_* variables should do a htons() in parse_arg() instead of ntohs() repeatedly in the bpf prog.

Reply:
Yes, it was missed before, it has been fixed, thank you!